innovation_sweet_spots/analysis/analysis_utils.py

```diff
@@ -75,58 +75,55 @@ def gtr_deduplicate_projects(gtr_docs: pd.DataFrame) -> pd.DataFrame:
     )[gtr_docs.columns]
 
 
-def gtr_funding_per_year(
-    gtr_docs: pd.DataFrame, min_year: int = None, max_year: int = None
+def gtr_funding_per_period(
+    gtr_docs: pd.DataFrame, period: str, min_year: int, max_year: int
 ) -> pd.DataFrame:
     """
-    Given a table with projects and their funding, return an aggregation by year
+    Given a table with projects and their funding, return an aggregation by period
 
     Args:
         gtr_docs: A dataframe with columns for 'start', 'project_id' and 'amount'
             (research funding) among other project data
-        min_year: Lower bound for years to keep
-        max_year: Higher bound for years to keep
+        period: Time period to group the data by, 'M', 'Q' or 'Y'
+        min_year: Earliest year to impute values for
+        max_year: Last year to impute values for
 
     Returns:
         A dataframe with the following columns:
-            'year',
-            'no_of_projects' - number of new projects in a given year,
-            'amount_total' - total amount of research funding in a given year,
-            'amount_median' - median project funding in a given year
+            'period' - time period
+            'no_of_projects' - number of new projects in a given period,
+            'amount_total' - total amount of research funding in a given period,
+            'amount_median' - median project funding in a given period
```

Comment:
Hi, looks like the function does not return `amount_median` anymore. Is that on purpose, or a bug?

Reply:
Hi Karlis , I removed it as I wasn't using it in the forecasting project and it didn't look it was used in the ISS analysis. Do you want it to stay in?